### PR TITLE
feat(performance): Queues module query functions

### PR DESCRIPTION
**Pull request by edwardgou-sentry**
Adds query hooks for getting queue metrics, transactions, and queue spans

### CODE REVIEW

File: static/app/views/performance/queues/queries/useQueuesMetricsQuery.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+type Props = {
+  destination?: string;
+  enabled?: boolean;
+  transaction?: string;
+};
+
+export function useQueuesMetricsQuery({destination, transaction, enabled}: Props) {
+  const mutableSearch = new MutableSearch(
+    'span.op:[queue.task.celery,queue.submit.celery]'
```
Comment (gggritso) [inline static/app/views/performance/queues/queries/useQueuesMetricsQuery.tsx:12]: This is starting to look like some kind of default conditions, right? If yes, worth dumping into `settings.tsx` for the module!

File: static/app/views/performance/queues/queries/useQueuesTimeSeriesQuery.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  results?.data?.['avg_if(span.self_time,span.op,queue.submit.celery)']?.data.forEach(
+    (interval, index) => {
+      yAxis.forEach(key => {
+        data[key].push({
+          value: results?.data?.[key]?.data[index].value ?? 0,
+          name: interval.name,
+        });
+      });
+    }
+  );
```
Comment (gggritso) [inline static/app/views/performance/queues/queries/useQueuesTimeSeriesQuery.tsx:43]: Could you add a note to explain why/what is happening here?

File: static/app/views/performance/queues/queries/useQueuesByTransactionQuery.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      'sum(span.self_time)',
+      'avg(span.self_time)',
+      'avg_if(span.self_time,span.op,queue.submit.celery)',
+      'avg_if(span.self_time,span.op,queue.task.celery)',
+    ],
+    enabled,
+    sorts: [],
+    limit: 10,
+    cursor,
+    referrer: 'api.starfish.queues-module-destination-summary',
```
Comment (gggritso) [inline static/app/views/performance/queues/queries/useQueuesByTransactionQuery.tsx:39]: We're past Starfish by now! IMO you can use the `api.performance.queues.` prefix for these. I know it's copypasta from the previous code, though, so no worries

**Review by gggritso — APPROVED** [review]

File: static/app/views/performance/queues/queries/useQueuesMetricsQuery.tsx (hunk 1 above)
Comment (edwardgou-sentry) [reply static/app/views/performance/queues/queries/useQueuesMetricsQuery.tsx]: good call, updated in `settings.tsx`!

File: static/app/views/performance/queues/queries/useQueuesTimeSeriesQuery.tsx (hunk 2 above)
Comment (edwardgou-sentry) [reply static/app/views/performance/queues/queries/useQueuesTimeSeriesQuery.tsx]: This just reduces the results to the interval array with less metadata. I don't actually need to do this, so I've removed this code!

**Pull request merged**